src/phases/query/CollScanComplexPredicate.yml

```diff
@@ -0,0 +1,521 @@
+SchemaVersion: 2018-07-01
+Owner: "@mongodb/query"
+Description: |
+  This workload tests the performance of collection scan queries with complex predicates of
+  various shapes (CNF, DNF, and mixed predicates with different levels of nestedness).
+
+GlobalDefaults:
+  Database: &Database {^Parameter: {Name: Database, Default: unused}}
+  DocumentCount: &DocumentCount {^Parameter: {Name: DocumentCount, Default: -1}}
+  Repeat: &Repeat {^Parameter: {Name: Repeat, Default: -1}}
+  Collection: &Collection Collection0
+  MaxPhases: &MaxPhases 20
+  
+  # The queries below are generated by selecting combinations of fields and comparison operations.
+  QueryFields: &QueryFields [
+    "a", 
+    "b", 
+    "c", 
+    "d", 
+    "f", 
+    "g", 
+    "h", 
+    "i",
+    "e.a", 
+    "e.b", 
+    "e.c", 
+    "e.d", 
+    "e.f", 
+    "e.g", 
+    "e.h", 
+    "e.i",
+    "nonexistent", 
+    "nullVal"
+  ]
+  QueryOps: &QueryOps ["$eq", "$lt", "$lte", "$gt", "$gte"]
+  
+  # Constructs an $and query with N $or clauses, where each $or contains 1-10 randomly generated
+  # comparisons.
+  CNFQuery: &CNFQuery 
```

Comment:
This could prove problematic for anyone investigating a regression in this workload, since it would be very difficult to derive the query being run here directly by looking at it (and consequently, very difficult to repro locally independent of genny). I see two options here.
1) obtain the query being generated once locally, and hard-code it directly in this file instead.
2) log the explain for this query on the command-line. there should be an actor that lets you do this (LoggingActor or similar). the downside is we will be collecting metrics for something we don't care about.

My vote would be for 1).

Reply:
I implemented 1) and hardcoded the queries in a separate file `CollScanComplexPredicateQueries.yml`.